web-admin/src/features/alerts/selectors.ts

```diff
@@ -21,6 +27,19 @@ export function useAlerts(instanceId: string, enabled = true) {
       query: {
         enabled: enabled && !!instanceId,
         refetchOnMount: true,
+        refetchInterval: (query) => {
+          const newInterval = calculateRefetchInterval(
+            currentRefetchInterval,
+            query.state.data,
+            query,
+          );
+          if (newInterval === false) {
+            currentRefetchInterval = INITIAL_REFETCH_INTERVAL;
+            return false;
+          }
+          currentRefetchInterval = newInterval;
+          return newInterval;
+        },
```

Comment:
Tracking state outside of the `calculateRefetchInterval` function makes this more complex than it feels like it should be. I'm wondering if there's a way `calculateRefetchInterval` could encompass all of the logic, so we could call it like this:
```
query: {
    ...
    refetchInterval: calculateRefetchInterval // more descriptive name could be something like `pollUntilResourcesReconciled`
}
```
One idea, could we add custom state to either `query.meta` or `query.state`, e.g. `query.state.currentRefetchInterval = X`?

Reply:
Agreed that we should keep the refetchInterval state encapsulated within the component.